kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/api/common/LocalRef.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.api.common;
+
+import java.util.Objects;
+
+/**
+ * Abstraction for references in one kubernetes resource to some kubernetes resource in the same namespace.
+ * Two LocalRefs are equal iff they have the same group, kind and name (they don't need to have the same class)
+ * @param <T> The Java type of the resource
+ */
+public abstract class LocalRef<T> {
```

Comment:
Sonar is squawking about T being unused, did you have some idea of a use for it?

Reply:
While it's true that `T` is not used in the class scope, I think it's still useful to be able to qualify a `LocalRef` with a Java type corresponding to the kind at external use sites. An example would be the signature `public static <T extends HasMetadata> LocalRef<T> toLocalRef(T ref)` in `ResourceUtil`